doc/dev/developer_guide/index.rst

```diff
@@ -791,6 +791,59 @@ the `cram task`_.
 .. _`cram`: https://bitheap.org/cram/
 .. _`cram task`: https://github.com/ceph/ceph/blob/master/qa/tasks/cram.py
 
+Unit testing of python modules
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Most python modules can be found under ``./src/pybind/``.
+
+If they have any python tests you can run them with the help of **tox**.
+**tox** itself is a generic virtualenv management and test command line tool
+
+To find out quickly if tox can be run you can either just try to run ``tox`` or find out if a
+``tox.ini`` exists.
+
+Currently the following modules use tox:
+
+- Ansible (``./src/pybind/mgr/ansible``)
+- Insights (``./src/pybind/mgr/insights``)
+- Orchestrator cli (``./src/pybind/mgr/orchestrator_cli``)
+- Manager core (``./src/pybind/mgr``)
+- Dashboard (``./src/pybind/mgr/dashboard``)
+- Python common (``./src/python-common/tox.ini``)
+
+Most tox configuration support multiple environments and tasks. You can see which environments and
+tasks are supported by looking into the ``tox.ini`` file to see what ``envlist`` is assigned.
+By just running ``tox`` in the directory where ``tox.ini`` lies, all environments will be run.
+But you can also run them with the ``run_tox.sh`` script which lies under ``./src/script``.
+
+Here some examples from the dashboard on how to specify different environments and run options::
+
+  ## Run Python 2+3 tests+lint commands:
+  $ ../../../script/run_tox.sh --tox-env py27,py3,lint,check
+  $ tox -e py27,py3,lint,check
+
+  ## Run Python 3 tests+lint commands:
+  $ ../../../script/run_tox.sh --tox-env py3,lint,check
+  $ tox -e py3,lint,check
+
+Manager core unit tests
+"""""""""""""""""""""""
+
+**tox** collects coverage information, to convert it into an HTML representation run::
+
+    $ coverage html
```

Comment:
how is "coverage" installed? might want to add a link to https://coverage.readthedocs.io/en/v4.5.x/install.html ? if i am reading this document, chances are that i don't understand what "coverage" is.